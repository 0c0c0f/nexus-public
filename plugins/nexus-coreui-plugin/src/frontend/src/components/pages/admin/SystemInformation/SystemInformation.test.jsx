/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {mount, shallow} from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';

import Axios from 'axios';
import SystemInformation from './SystemInformation';
import { Button, Checkbox, Textfield, Select } from 'nexus-ui-plugin';
import UIStrings from '../../../../constants/UIStrings';

jest.mock('axios', () => {  // Mock response from axios
  return {
    ...jest.requireActual('axios'), // Use most functions from actual axios
    get: jest.fn((url) => Promise.resolve({
      data: {
        'nexus-status': {
          'status': 'value'
        },
        'nexus-node': {
          'node': 0
        },
        'nexus-configuration': {
          'enabled': true
        },
        'nexus-properties': {
          'property': false
        },
        'nexus-license': {
          'fingerprint': 'hash'
        },
        'nexus-bundles': {
          '0': {
            'bundleId': 0,
            'enabled': true,
            'config': 'value'
          }
        },
        'system-time': {
          'time': 'value'
        },
        'system-properties': {
          'property': 'value'
        },
        'system-environment': {
          'property': 'value'
        },
        'system-runtime': {
          'property': 'value'
        },
        'system-network': {
          'en0': {
            'enabled': true
          }
        },
        'system-filestores': {
          '/dev/null': {
            'path': '/dev/null'
          }
        }
      }
    }))
  };
});

describe('SystemInformation', () => {
  // see https://github.com/airbnb/enzyme/issues/1587
  const waitForDataFromApi = (wrapper) => act(() => Promise.resolve(wrapper)
      .then(() => wrapper.update())
      .then(() => wrapper.update())
  );

  it('renders correctly', async () => {
    const wrapper = mount(<SystemInformation/>);

    await waitForDataFromApi(wrapper);

    expect(wrapper).toMatchSnapshot();
  });
});
