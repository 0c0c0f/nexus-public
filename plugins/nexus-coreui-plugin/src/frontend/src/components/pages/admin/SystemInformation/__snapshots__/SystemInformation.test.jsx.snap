// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemInformation renders correctly 1`] = `
<SystemInformation>
  <BreadcrumbActions>
    <div
      className="nxrm-breadcrumb--actions"
    >
      <Button
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        <button
          className="nxrm-button nxrm-button-primary"
          disabled={false}
          onClick={[Function]}
          type="input"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f019",
                  "M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z",
                ],
                "iconName": "download",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull="left"
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-download fa-w-16 fa-pull-left "
              data-icon="download"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
          Download as JSON
        </button>
      </Button>
    </div>
  </BreadcrumbActions>
  <ContentBody
    className="nxrm-system-information"
  >
    <div
      className="nxrm-content-body nxrm-system-information"
    >
      <SystemInformationSection
        information={
          Object {
            "status": "value",
          }
        }
        sectionName="nexus-status"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              nexus-status
            </h2>
            <Information
              information={
                Object {
                  "status": "value",
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="status"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          status
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          value
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "node": 0,
          }
        }
        sectionName="nexus-node"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              nexus-node
            </h2>
            <Information
              information={
                Object {
                  "node": 0,
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="node"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          node
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          0
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "enabled": true,
          }
        }
        sectionName="nexus-configuration"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              nexus-configuration
            </h2>
            <Information
              information={
                Object {
                  "enabled": true,
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="enabled"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          enabled
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          true
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "property": false,
          }
        }
        sectionName="nexus-properties"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              nexus-properties
            </h2>
            <Information
              information={
                Object {
                  "property": false,
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="property"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          property
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          false
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "fingerprint": "hash",
          }
        }
        sectionName="nexus-license"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              nexus-license
            </h2>
            <Information
              information={
                Object {
                  "fingerprint": "hash",
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="fingerprint"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          fingerprint
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          hash
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <NestedSystemInformationSection
        sectionInformation={
          Object {
            "0": Object {
              "bundleId": 0,
              "config": "value",
              "enabled": true,
            },
          }
        }
        sectionName="nexus-bundles"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              nexus-bundles
            </h2>
            <div
              key="0"
            >
              <h3>
                0
              </h3>
              <Information
                information={
                  Object {
                    "bundleId": 0,
                    "config": "value",
                    "enabled": true,
                  }
                }
              >
                <table
                  className="nxrm-information"
                >
                  <tbody>
                    <InformationRow
                      key="bundleId"
                    >
                      <tr
                        className="nxrm-information--row"
                      >
                        <InformationName>
                          <td
                            className="nxrm-information--name"
                          >
                            bundleId
                          </td>
                        </InformationName>
                        <InformationValue>
                          <td
                            className="nxrm-information--value"
                          >
                            0
                          </td>
                        </InformationValue>
                      </tr>
                    </InformationRow>
                    <InformationRow
                      key="enabled"
                    >
                      <tr
                        className="nxrm-information--row"
                      >
                        <InformationName>
                          <td
                            className="nxrm-information--name"
                          >
                            enabled
                          </td>
                        </InformationName>
                        <InformationValue>
                          <td
                            className="nxrm-information--value"
                          >
                            true
                          </td>
                        </InformationValue>
                      </tr>
                    </InformationRow>
                    <InformationRow
                      key="config"
                    >
                      <tr
                        className="nxrm-information--row"
                      >
                        <InformationName>
                          <td
                            className="nxrm-information--name"
                          >
                            config
                          </td>
                        </InformationName>
                        <InformationValue>
                          <td
                            className="nxrm-information--value"
                          >
                            value
                          </td>
                        </InformationValue>
                      </tr>
                    </InformationRow>
                  </tbody>
                </table>
              </Information>
            </div>
          </div>
        </Section>
      </NestedSystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "time": "value",
          }
        }
        sectionName="system-time"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              system-time
            </h2>
            <Information
              information={
                Object {
                  "time": "value",
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="time"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          time
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          value
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "property": "value",
          }
        }
        sectionName="system-properties"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              system-properties
            </h2>
            <Information
              information={
                Object {
                  "property": "value",
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="property"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          property
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          value
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "property": "value",
          }
        }
        sectionName="system-environment"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              system-environment
            </h2>
            <Information
              information={
                Object {
                  "property": "value",
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="property"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          property
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          value
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <SystemInformationSection
        information={
          Object {
            "property": "value",
          }
        }
        sectionName="system-runtime"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              system-runtime
            </h2>
            <Information
              information={
                Object {
                  "property": "value",
                }
              }
            >
              <table
                className="nxrm-information"
              >
                <tbody>
                  <InformationRow
                    key="property"
                  >
                    <tr
                      className="nxrm-information--row"
                    >
                      <InformationName>
                        <td
                          className="nxrm-information--name"
                        >
                          property
                        </td>
                      </InformationName>
                      <InformationValue>
                        <td
                          className="nxrm-information--value"
                        >
                          value
                        </td>
                      </InformationValue>
                    </tr>
                  </InformationRow>
                </tbody>
              </table>
            </Information>
          </div>
        </Section>
      </SystemInformationSection>
      <NestedSystemInformationSection
        sectionInformation={
          Object {
            "en0": Object {
              "enabled": true,
            },
          }
        }
        sectionName="system-network"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              system-network
            </h2>
            <div
              key="en0"
            >
              <h3>
                en0
              </h3>
              <Information
                information={
                  Object {
                    "enabled": true,
                  }
                }
              >
                <table
                  className="nxrm-information"
                >
                  <tbody>
                    <InformationRow
                      key="enabled"
                    >
                      <tr
                        className="nxrm-information--row"
                      >
                        <InformationName>
                          <td
                            className="nxrm-information--name"
                          >
                            enabled
                          </td>
                        </InformationName>
                        <InformationValue>
                          <td
                            className="nxrm-information--value"
                          >
                            true
                          </td>
                        </InformationValue>
                      </tr>
                    </InformationRow>
                  </tbody>
                </table>
              </Information>
            </div>
          </div>
        </Section>
      </NestedSystemInformationSection>
      <NestedSystemInformationSection
        sectionInformation={
          Object {
            "/dev/null": Object {
              "path": "/dev/null",
            },
          }
        }
        sectionName="system-filestores"
      >
        <Section>
          <div
            className="nxrm-section"
          >
            <h2>
              system-filestores
            </h2>
            <div
              key="/dev/null"
            >
              <h3>
                /dev/null
              </h3>
              <Information
                information={
                  Object {
                    "path": "/dev/null",
                  }
                }
              >
                <table
                  className="nxrm-information"
                >
                  <tbody>
                    <InformationRow
                      key="path"
                    >
                      <tr
                        className="nxrm-information--row"
                      >
                        <InformationName>
                          <td
                            className="nxrm-information--name"
                          >
                            path
                          </td>
                        </InformationName>
                        <InformationValue>
                          <td
                            className="nxrm-information--value"
                          >
                            /dev/null
                          </td>
                        </InformationValue>
                      </tr>
                    </InformationRow>
                  </tbody>
                </table>
              </Information>
            </div>
          </div>
        </Section>
      </NestedSystemInformationSection>
    </div>
  </ContentBody>
</SystemInformation>
`;
